aya/src/bpf.rs

```diff
@@ -224,6 +228,62 @@ impl<'a> EbpfLoader<'a> {
         self
     }
 
+    /// Allows programs containing unsupported maps for the current kernel to be loaded
+    /// by skipping map creation and relocation before loading.
+    ///
+    /// This is useful when you have a single ebpf program containing e.g. a `RingBuf`
+    /// and a `PerfEventArray` and you decide which one to use before loading the bytecode.
+    ///
+    /// Must be used with `.set_global()` to signal if the map is supported in the ebpf program.
+    ///
+    /// # Example
+    ///
+    /// ```no_run
+    /// use aya::EbpfLoader;
+    /// use aya_obj::generated::bpf_map_type;
+    /// use std::collections::HashSet;
+    ///
+    /// let ringbuf_supported = 0;
+    /// let mut set = HashSet::new();
+    /// set.insert(bpf_map_type::BPF_MAP_TYPE_RINGBUF);
+    /// let ebpf = EbpfLoader::new()
+    ///     .ignore_maps_by_type(set)
+    ///     .set_global("RINGBUF_SUPPORTED", &ringbuf_supported, true)
+    ///     .load_file("file.o")?;
+    /// # Ok::<(), aya::EbpfError>(())
+    /// ```
+    ///
+    pub fn ignore_maps_by_type(&mut self, set: HashSet<bpf_map_type>) -> &mut Self {
+        self.ignore_maps_by_type = set;
+        self
+    }
+
+    /// Allows programs containing unsupported maps for the current kernel to be loaded
```

Comment:
```
Don't load maps with the given names.

Similar to `ignore_maps_by_type()` but skips based on map name instead of type.
```

No need to repeat the same example. Or alternatively provide a different example
(for inspiration: something I plan to do with this is to avoid loading the MEMORY_EVENTS map of
my profiler when memory profiling is disabled)